draw/src/shader/draw_text.rs

```diff
@@ -661,9 +661,21 @@ impl DrawText {
                         let mut iter = glyph_infos.iter().peekable();
                         while let Some(glyph_info) = iter.next() {
                             let glyph_start = start + glyph_info.cluster;
-                            let glyph_end = start + iter.peek().map_or(string.len(), |glyph_info| glyph_info.cluster);
+
+                            let glyph_end = if let Some(next_info) = iter.peek() {
+                                start + next_info.cluster
+                            } else {
+                                // If it is the last glyph, use the length of the entire string
+                                start + string.len()
+                            };
+                            // Add safety checks.
+                            if glyph_end <= glyph_start || glyph_start >= text.len() {
+                                continue;
+                            }
+                            // Use safe maximum boundaries
+                            let safe_end = glyph_end.min(text.len());
```

Comment:
This check also should not be necessary, for the same reasons as outlined above.